### DISCUSSION TITLE
how to config task more friendly?

### DISCUSSION THREAD

**Discussion by conderls** [SETL-Framework/setl#227] [Q&A] [answered]
SETL framework basically features modularizing data transformation logic and speeding up your development based on Apache Spark. It seems inconvenient to update parameters after the Uber-Jar packaged, which is time-consuming.

So, what's the better way to update/pass parameters for SETL tasks without update config file and recompile the package.

Normally, parameters are passed to main class entry, then parse the args. 

With SETL, it is convenient and clear to put all arguments in the configure file with comments, but inconvenient to update, especially for spark tasks which may need update params for Executor and Driver at the same time.

**Comment by qxzzxq** [comment, marked as answer]
Sorry for the late reply. You can use environment variables for parameters that need to be updated. 

For example, in the configuration file:
```
myvariable = ${MY_VAR}
```
Then `myvariable` will be set to the value of the environment variable `MY_VAR`.

For more features, please refer to https://github.com/lightbend/config
